rust/oak/src/io/mod.rs

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2018 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Wrappers for Oak SDK types to allow their use with std::io.
+
+use crate::{channel_write, OakStatus};
+use std::io;
+
+/// Wrapper for a WriteHandle to implement the [`std::io::Write`] trait.
+///
+/// Methods taking `Write` trait objects require mutable references,
```

Comment:
Not sure I understand why `WriteHandle` is not sufficient?

Reply:
It just felt &hellip; untidy.  The `Handle` types are `Copy` / POD types and so generally passed by value, whereas things using `Write` generally want a mutable reference; also, pulling the wrapper into an `io` submodule meant that all of the `std:io` stuff is in one place, and `oak/lib.rs` entirely deals with `OakStatus` values.  Happy to drop the `io` submodule if you feel strongly about it though.